### PR TITLE
Validation for tag keys and values

### PR DESCRIPTION
**Pull request by ezequielramos**
- Invalid values on tags were creating an invalid message to graphite. Added a validation similar to metric name to validate whitespaces.
- Added tests to simulate that exception.

### CODE REVIEW

File: graphyte.py [hunk 1]
```diff
@@ -25,6 +25,12 @@
 default_sender = None
 logger = logging.getLogger(__name__)
 
+def has_whitespace(value):
```
Comment (benhoyt) [inline graphyte.py]: Can we change this to `_has_whitespace` to indicate it's private?

File: graphyte.py [hunk 2]
```diff
@@ -25,6 +25,12 @@
...
+    if type(value) == int:
```
Comment (benhoyt) [inline graphyte.py]: Exact type checks aren't very Pythonic ("if it walks like a duck and quacks like a duck"). Let's change this to `if not isinstance(value, str)` and drop the `if type(value) == str` test altogether.

File: graphyte.py [hunk 3]
```diff
@@ -66,13 +72,19 @@ def stop(self):
...
         """Build a Graphite message to send and return it as a byte string."""
-        if not metric or metric.split(None, 1)[0] != metric:
+        if has_whitespace(metric):
             raise ValueError('"metric" must not have whitespace in it')
         if not isinstance(value, (int, float)):
             raise TypeError('"value" must be an int or a float, not a {}'.format(
                 type(value).__name__))
 
-        tags_suffix = ''.join(';{}={}'.format(x[0], x[1]) for x in sorted(tags.items()))
+        tags_suffix = ''
```
Comment (benhoyt) [inline graphyte.py]: Using str `+=` in Python can lead to O(N^2) behavior, so better to build a list and join. Hopefully no one has that many tags, but might as well use best practices. So maybe:

```
tags_strs = [';{}={}'.format(k, v) for k, v in sorted(tags.items())]
if any(has_whitespace(t) for t in tags_strs):
    raise ValueError('"tags" keys and values must not have whitespace in them')
tags_suffix = ''.join(tags_strs)
```

**Review by benhoyt — CHANGES_REQUESTED** [review]
Good idea, thanks! Added a few review comments.

File: graphyte.py (hunk 1 above)
Comment (ezequielramos) [reply graphyte.py]: Sure! Done

File: graphyte.py (hunk 2 above)
Comment (ezequielramos) [reply graphyte.py]: Nice tip!
Well, actually I'm casting everything to _unicode_ before calling __has_whitespace_ so these tests aren't really necessary anymore.

File: graphyte.py (hunk 3 above)
Comment (ezequielramos) [reply graphyte.py]: Oh, didn't think of that!
I change it to this better approach anyway! Just made a change to cast to _unicode_. 
`tags_strs = [u';{}={}'.format(k, v) for k, v in sorted(tags.items())]`

**Review by ezequielramos — COMMENTED** [review]
Done.
Also added a new test for tags keys/values on unicode format.
